x-pack/plugins/observability/server/lib/rules/slo_burn_rate/executor.ts

```diff
@@ -204,7 +255,8 @@ export const getRuleExecutor = ({
       const { getRecoveredAlerts } = alertFactory.done();
       const recoveredAlerts = getRecoveredAlerts();
       for (const recoveredAlert of recoveredAlerts) {
-        const alertId = `alert-${slo.id}-${slo.revision}`;
+        const revision = sloSchema.is(slo) ? slo.revision : 1;
+        const alertId = `alert-${slo.id}-${revision}`;
```

Comment:
Could we just fallback to 1 if revision is undefined? So we don't rerun a runtime schema validation again:
```suggestion
        const alertId = `alert-${slo.id}-${slo.revision ?? 1}`;
```